bezier-rs/lib/src/lib.rs

```diff
@@ -305,4 +305,68 @@ impl Bezier {
 		};
 		bezier_starting_at_t1.split(adjusted_t2)[t2_split_side]
 	}
+
+	/// Find the roots of the linear equation ax + b
+	fn solve_linear(a: f64, b: f64) -> Vec<f64> {
+		let mut roots = Vec::new();
+		if b != 0. {
+			roots.push(-b / a);
+		}
+		roots
+	}
+
+	/// Find the roots of the linear equation ax^2 + bx + c
+	fn solve_quadratic(discriminant: f64, two_times_a: f64, b: f64, c: f64) -> Vec<f64> {
+		let mut roots = Vec::new();
+		println!("{}", discriminant);
+		if two_times_a != 0. {
+			// println!("Solving Quadratic");
+			if discriminant > 0. {
+				let root_discriminant = discriminant.sqrt();
+				roots.push((-b + root_discriminant) / (two_times_a));
+				roots.push((-b - root_discriminant) / (two_times_a));
+			} else if discriminant == 0. {
+				roots.push(-b / (two_times_a));
+			}
+			println!("{:#?}", roots);
+		} else {
+			roots = Bezier::solve_linear(b, c);
+		}
+		roots
+	}
+
+	/// Returns two lists of `t`-values representing the local extrema of the `x` and `y` parametric curves respectively
+	/// The local extrema are defined to be points at which the derivative of the curve is equal to zero
+	fn _local_extrema(&self) -> [Vec<f64>; 2] {
+		match self.handles {
+			BezierHandles::Quadratic { handle } => {
+				let a = handle - self.start;
+				let b = self.end - handle;
+				let b_minus_a = b - a;
+				[Bezier::solve_linear(a.x, b_minus_a.x), Bezier::solve_linear(a.y, b_minus_a.y)]
+			}
+			BezierHandles::Cubic { handle_start, handle_end } => {
+				let a = 3. * (-self.start + 3. * handle_start - 3. * handle_end + self.end);
+				let b = 6. * (self.start - 2. * handle_start + handle_end);
+				let c = 3. * (handle_start - self.start);
+				let discriminant = b * b - 4. * a * c;
+				let two_times_a = 2. * a;
+				[
+					Bezier::solve_quadratic(discriminant.x, two_times_a.x, b.x, c.x),
+					Bezier::solve_quadratic(discriminant.y, two_times_a.y, b.y, c.y),
+				]
+			}
+		}
+	}
+
+	/// Returns two lists of `t`-values representing the local extrema of the `x` and `y` parametric curves respectively
+	/// The extrema returned are restricted to those which fall within `[0, 1]`
```

Comment:
Could make the comment more descriptive like "points are filtered such that they fall within the range [0,1]".